lib/xenops_server.ml

```diff
@@ -2153,8 +2153,18 @@ and perform_exn ?subtask ?result (op : operation) (t : Xenops_task.task_handle)
             let final_handshake () =
               Handshake.send ~verbose:true mem_fd Handshake.Success ;
               debug "VM.migrate: Synchronisation point 3" ;
-              Handshake.recv_success mem_fd ;
-              debug "VM.migrate: Synchronisation point 4"
+              match Handshake.recv mem_fd with
+              | Success ->
+                  debug "VM.migrate: Synchronisation point 4"
+              | Error msg ->
+                  (* at this point, the VM has already been transferred to
+                     the destination host. even though the destination host
+                     failed to respond successfully to our handshake, the VM
+                     should still be running correctly *)
+                  error
+                    "VM.migrate: Failed during Synchronisation point 4. msg: %s"
+                    msg ;
+                  raise (Xenopsd_error (Internal_error msg))
```

Comment:
I don't think that this actually changes anything, besides the error the is returned, correct? Originally, `Handshake.recv_success` would have also raised an exception, but the following: `raise (Remote_failed ("error from remote: " ^ x))`.

Reply:
Indeed, I suspect it doesn't change anything (I changed it because I experimented with putting VM_remove here), but I prefer to be explicit here